data-prepper-expression/src/main/antlr/DataPrepperExpression.g4

```diff
@@ -141,20 +141,18 @@ literal
 
 
 Integer
-    : '0'
+    : ZERO
     | NonZeroDigit Digit*
     ;
 
 Float
-    : NonZeroDigit? Digit '.' Digit
-    | NonZeroDigit? Digit '.' Digit+ NonZeroDigit
-    | '.' Digit
-    | '.' Digit* NonZeroDigit
+    : Integer '.' ZERO* Integer
+    | Integer '.' Integer EXPONENTLETTER SUBTRACT? Integer
```

Comment:
Should it be like this? So numbers like `123.00045e6` would be allowed?
```suggestion
    | Integer '.' ZERO* Integer EXPONENTLETTER SUBTRACT? Integer
```

Reply:
I guess we need ZERO* in both cases. Will update the diffs.